app.go

```diff
@@ -787,7 +788,15 @@ func (app *App) Stop(ctx context.Context) (err error) {
 // using the Shutdown functionality (see the Shutdowner documentation for details).
 func (app *App) Done() <-chan os.Signal {
 	c := make(chan os.Signal, 1)
-	signal.Notify(c, _sigINT, _sigTERM)
+
+	// If shutdown signal has been received already
+	// send it else wait for user to send a termination
+	// signal.
+	if app.shutdownSig != nil {
+		c <- app.shutdownSig
+	} else {
+		signal.Notify(c, _sigINT, _sigTERM)
+	}
```

Comment:
This isn't thread safe. Someone might call Shutdown (which calls broadcastSignal) in another goroutine at the same time this is happening. This needs to be done after we have a lock.